### PR TITLE
Nerfs 049 grab

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

049 grab is too strong - this weakens him.

Also GOD DAMN IT EGOR! I HATE YOU!!!!!

## Why It's Good For The Game

See: 15 people on discord complaining about 049 being OP.

## Changelog

:cl:
qol: Actions failed due to action cooldown give a balloon alert instead of a to_chat message.
qol: 049 grab uses balloon alerts.
balance: Instead of dealing a flat 10 weakness every step, 049 curing now gradually increases in power. The first step has no effect, the second gives 3 stacks of weakness, the third gives 8 stacks, and the fourth and final one gives 12 stacks of weakness, as well as 4 stacks of paralysis.
balance: 049 curing grab has a delay of 10 seconds (from 4 seconds).
balance: 049 now has partial flash resistance (from FULL flash resistance).
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
why do we even have an SCP specific grab...
oh
Egor
that's why

**Comment by TheDarkElites** [comment]
also, calling it grab SCP might become problematic if we have another SCP that can grab (like 096, for example). You might wanna rename it to something like 049 grabs.

**Comment by aidan-bowman** [comment]
> also, calling it grab SCP might become problematic if we have another SCP that can grab (like 096, for example). You might wanna rename it to something like 049 grabs.

done, btw

File: code/modules/mob/grab/scp/grab_049.dm [hunk 1]
```diff
@@ -12,20 +12,18 @@
...
+	stop_move = TRUE
+	can_grab_self = FALSE
+
+	action_cooldown = 10 SECONDS
+
+	/// When TRUE - blocks quick_stabs() proc from continuing
 	var/stop_effects = FALSE
 
-// This causes the user to temporarily stun and weaken the target, dealing pain and slight brute damage
-/datum/grab/plague_doctor/proc/AttemptCure(obj/item/grab/normal/G)
```
Comment (Tsurupeta) [inline code/modules/mob/grab/scp/grab_049.dm:28]: bruh

File: code/modules/mob/grab/scp/grab_049.dm [hunk 2]
```diff
@@ -45,33 +43,36 @@
...
 			if(4)
+				target.Weaken(12)
+				target.Paralyse(4)
 				to_chat(user, SPAN_NOTICE("You spend a great deal of time expertly curing this victim's disease."))
 				user.visible_message(SPAN_DANGER("[user] begins performing a horrifying procedure on [target]!"))
 				playsound(target, 'sounds/weapons/bladeslice.ogg', 50, TRUE, 7)
 
 		if(!do_after(user, 10 SECONDS, target))
-			to_chat(user, SPAN_WARNING("The procedure has been interrupted!"))
+			user.balloon_alert_to_viewers("curing interuptted!")
```
Comment (Tsurupeta) [inline code/modules/mob/grab/scp/grab_049.dm:75]: why viewers??????

File: code/modules/mob/grab/scp/grab_049.dm (hunk 1 above)
Comment (Tsurupeta) [reply code/modules/mob/grab/scp/grab_049.dm:28]: Dont rename the proc, we love PascalCase now

File: code/modules/mob/grab/scp/grab_049.dm [hunk 3]
```diff
@@ -81,8 +82,8 @@
 	admin_attack_log(user, target, "'Cured' their victim.", "Was 'cured'.", "'cured'")
 	qdel(G)
 
-// Splashes neat effect all around the target
-/datum/grab/plague_doctor/proc/DoEffects(obj/item/grab/normal/G)
+/// This proc repeatedly does minor Brute damage to the target with a few visual effects (bloodspatters, random noises, etc)
+/datum/grab/plague_doctor/proc/quick_stabs(obj/item/grab/normal/G)
```
Comment (Tsurupeta) [inline code/modules/mob/grab/scp/grab_049.dm]: Same here, pascalcase

**Review by Tsurupeta — CHANGES_REQUESTED** [review]

File: code/modules/mob/grab/scp/grab_049.dm (hunk 2 above)
Comment (aidan-bowman) [reply code/modules/mob/grab/scp/grab_049.dm:75]: it's relevant to other players as well

File: code/modules/mob/grab/scp/grab_049.dm (hunk 1 above)
Comment (aidan-bowman) [reply code/modules/mob/grab/scp/grab_049.dm:28]: fuck pascalcase, snake case my beloved

File: code/modules/mob/grab/scp/grab_049.dm (hunk 2 above)
Comment (Tsurupeta) [reply code/modules/mob/grab/scp/grab_049.dm:75]: Ok, then I'm closing this PR.

File: code/modules/mob/grab/scp/grab_049.dm (hunk 1 above)
Comment (Tsurupeta) [reply code/modules/mob/grab/scp/grab_049.dm:28]: Ok then I'm closing this

**Comment by aidan-bowman** [comment]
ok bud, i'll make it pascalcase but ONLY because you asked nicely

**Pull request merged**